### ISSUE TITLE
Modify System Tests used Docker image for environment provisioning

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/wazuh-qa#4592]
## Description
During https://github.com/wazuh/wazuh/issues/13373, the System tests are required to be executed to detect possible flaws in the upcoming Python 3.10 upgrade. It was found that the provisioning phase of these tests is failing for the Wazuh branch `4.7.0` and upwards.

### Current behavior

The base Docker image (`dontpanicat/debian:buster`) used for the provisioning phase contains a GCC (C/C++ compiler) version (`8.3.0`) that does not work for the `4.7.0` nor `master` branch code making the provisioning phase to fail with the following output:
<details><summary>Provisioning error</summary>

    ...
    4- Installing the system
    DIR="/var/ossec"
     - Running the Makefile
    Makefile:2510: warning: overriding recipe for target 'win32/ui_resource.o'
    Makefile:2450: warning: ignoring old recipe for target 'win32/ui_resource.o'
    Makefile:2513: warning: overriding recipe for target 'win32/auth_resource.o'
    Makefile:2453: warning: ignoring old recipe for target 'win32/auth_resource.o'
    cp /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libstdc++.so.6 libstdc++.so.6
    cp /usr/lib/gcc/x86_64-linux-gnu/8/libgcc_s.so.1 libgcc_s.so.1
    strip -x libgcc_s.so.1
    strip -x libstdc++.so.6
    make build_sysinfo build_shared_modules build_syscollector
    make[1]: Entering directory '/wazuh/src'
    Makefile:2510: warning: overriding recipe for target 'win32/ui_resource.o'
    Makefile:2450: warning: ignoring old recipe for target 'win32/ui_resource.o'
    Makefile:2513: warning: overriding recipe for target 'win32/auth_resource.o'
    Makefile:2453: warning: ignoring old recipe for target 'win32/auth_resource.o'
    cd data_provider/ && mkdir -p build && cd build && cmake     .. && make
    cd shared_modules/dbsync/ && mkdir -p build && cd build && cmake    .. && make
    -- The C compiler identification is GNU 8.3.0
    -- The C compiler identification is GNU 8.3.0
    -- The CXX compiler identification is GNU 8.3.0
    -- The CXX compiler identification is GNU 8.3.0
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Detecting C compile features
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Detecting C compile features - done
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Detecting CXX compile features - done
    -- Performing Test HAS_OPTIONAL
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /wazuh/src/shared_modules/dbsync/build
    make[2]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    make[3]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    Scanning dependencies of target dbsync
    make[4]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    [ 40%] Building CXX object CMakeFiles/dbsync.dir/src/dbsync_implementation.cpp.o
    [ 40%] Building CXX object CMakeFiles/dbsync.dir/src/dbsyncPipelineFactory.cpp.o
    [ 40%] Building CXX object CMakeFiles/dbsync.dir/src/dbsync.cpp.o
    [ 40%] Building CXX object CMakeFiles/dbsync.dir/src/sqlite/sqlite_dbengine.cpp.o
    [ 50%] Building CXX object CMakeFiles/dbsync.dir/src/sqlite/sqlite_wrapper.cpp.o
    -- Performing Test HAS_OPTIONAL - Success
    -- Performing Test HAS_FILESYSTEM
    -- Performing Test HAS_FILESYSTEM - Success
    ==============================================
    Downloading benchmark...
    Extracting benchmark.tar.gz
    Cleaning benchmark.tar.gz
    ==============================================
    Downloading cpp-httplib...
    Extracting cpp-httplib.tar.gz
    Cleaning cpp-httplib.tar.gz
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /wazuh/src/data_provider/build
    make[2]: Entering directory '/wazuh/src/data_provider/build'
    make[3]: Entering directory '/wazuh/src/data_provider/build'
    make[4]: Entering directory '/wazuh/src/data_provider/build'
    Scanning dependencies of target urlrequest
    make[4]: Leaving directory '/wazuh/src/data_provider/build'
    make[4]: Entering directory '/wazuh/src/data_provider/build'
    [  5%] Building CXX object /wazuh/src/shared_modules/http-request/build/CMakeFiles/urlrequest.dir/src/UNIXSocketRequest.cpp.o
    [ 11%] Building CXX object /wazuh/src/shared_modules/http-request/build/CMakeFiles/urlrequest.dir/src/HTTPRequest.cpp.o
    In file included from /wazuh/src/shared_modules/http-request/src/factoryRequestImplemetator.hpp:15,
                     from /wazuh/src/shared_modules/http-request/src/HTTPRequest.cpp:13:
    /wazuh/src/shared_modules/http-request/src/IRequestImplementator.hpp:73:38: warning: inline function 'virtual const string IRequestImplementator::response()' used but never defined
         virtual inline const std::string response() = 0;
                                          ^~~~~~~~
    In file included from /wazuh/src/shared_modules/http-request/src/factoryRequestImplemetator.hpp:15,
                     from /wazuh/src/shared_modules/http-request/src/UNIXSocketRequest.cpp:13:
    /wazuh/src/shared_modules/http-request/src/IRequestImplementator.hpp:73:38: warning: inline function 'virtual const string IRequestImplementator::response()' used but never defined
         virtual inline const std::string response() = 0;
                                          ^~~~~~~~
    [ 16%] Linking CXX static library /wazuh/src/data_provider/build/lib/liburlrequest.a
    make[4]: Leaving directory '/wazuh/src/data_provider/build'
    [ 16%] Built target urlrequest
    make[4]: Entering directory '/wazuh/src/data_provider/build'
    Scanning dependencies of target sysinfo
    make[4]: Leaving directory '/wazuh/src/data_provider/build'
    make[4]: Entering directory '/wazuh/src/data_provider/build'
    [ 33%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/packageLinuxParserApk.cpp.o
    [ 33%] Building CXX object CMakeFiles/sysinfo.dir/src/network/networkInterfaceLinux.cpp.o
    [ 33%] Building CXX object CMakeFiles/sysinfo.dir/src/osinfo/sysOsParsers.cpp.o
    [ 38%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/packageLinuxParserExtra.cpp.o
    [ 44%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/packageLinuxParserRpm.cpp.o
    [ 50%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/packageLinuxParserDeb.cpp.o
    [ 55%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/packageLinuxParserRpmLegacy.cpp.o
    [ 60%] Linking CXX shared library lib/libdbsync.so
    make[4]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    [ 60%] Built target dbsync
    make[4]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    Scanning dependencies of target dbsync_example
    make[4]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    [ 70%] Building CXX object example/CMakeFiles/dbsync_example.dir/main.cpp.o
    make[4]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    Scanning dependencies of target dbsync_test_tool
    make[4]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/dbsync/build'
    [ 80%] Building CXX object testtool/CMakeFiles/dbsync_test_tool.dir/main.cpp.o
    [ 61%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/packageLinuxParserSnap.cpp.o
    [ 90%] Linking CXX executable ../bin/dbsync_example
    make[4]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    [ 90%] Built target dbsync_example
    [ 66%] Building CXX object CMakeFiles/sysinfo.dir/src/packages/rpmPackageManager.cpp.o
    [ 77%] Building CXX object CMakeFiles/sysinfo.dir/src/sysInfoLinux.cpp.o
    [ 77%] Building CXX object CMakeFiles/sysinfo.dir/src/utilsWrapperLinux.cpp.o
    [ 83%] Building CXX object CMakeFiles/sysinfo.dir/src/sysInfo.cpp.o
    In file included from /wazuh/src/data_provider/src/sysInfoLinux.cpp:15:
    /wazuh/src/data_provider/src/packages/modernPackageDataRetriever.hpp: In static member function 'static void ModernFactoryPackagesCreator<true>::getPackages(const std::map<std::__cxx11::basic_string<char>, std::set<std::__cxx11::basic_string<char> > >&, std::function<void(nlohmann::json_abi_v3_11_2::basic_json<>&)>)':
    /wazuh/src/data_provider/src/packages/modernPackageDataRetriever.hpp:53:18: error: cannot deduce template arguments for 'PYPI' from ()
                 PYPI().getPackages(paths.at("PYPI"), callback);
                      ^
    /wazuh/src/data_provider/src/packages/modernPackageDataRetriever.hpp:54:17: error: cannot deduce template arguments for 'NPM' from ()
                 NPM().getPackages(paths.at("NPM"), callback);
                     ^
    make[4]: *** [CMakeFiles/sysinfo.dir/build.make:199: CMakeFiles/sysinfo.dir/src/sysInfoLinux.cpp.o] Error 1
    make[4]: *** Waiting for unfinished jobs....
    make[4]: Leaving directory '/wazuh/src/data_provider/build'
    make[3]: *** [CMakeFiles/Makefile2:136: CMakeFiles/sysinfo.dir/all] Error 2
    make[3]: Leaving directory '/wazuh/src/data_provider/build'
    make[2]: *** [Makefile:114: all] Error 2
    make[2]: Leaving directory '/wazuh/src/data_provider/build'
    make[1]: *** [Makefile:1674: build_sysinfo] Error 2
    make[1]: *** Waiting for unfinished jobs....
    [100%] Linking CXX executable ../bin/dbsync_test_tool
    make[4]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    [100%] Built target dbsync_test_tool
    make[3]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    make[2]: Leaving directory '/wazuh/src/shared_modules/dbsync/build'
    cd shared_modules/rsync/ && mkdir -p build && cd build && cmake     .. && make
    -- The C compiler identification is GNU 8.3.0
    -- The CXX compiler identification is GNU 8.3.0
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /wazuh/src/shared_modules/rsync/build
    make[2]: Entering directory '/wazuh/src/shared_modules/rsync/build'
    make[3]: Entering directory '/wazuh/src/shared_modules/rsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/rsync/build'
    Scanning dependencies of target rsync
    make[4]: Leaving directory '/wazuh/src/shared_modules/rsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/rsync/build'
    [ 25%] Building CXX object CMakeFiles/rsync.dir/src/rsyncImplementation.cpp.o
    [ 25%] Building CXX object CMakeFiles/rsync.dir/src/rsync.cpp.o
    [ 37%] Linking CXX shared library lib/librsync.so
    make[4]: Leaving directory '/wazuh/src/shared_modules/rsync/build'
    [ 37%] Built target rsync
    make[4]: Entering directory '/wazuh/src/shared_modules/rsync/build'
    Scanning dependencies of target rsync_test_tool
    make[4]: Leaving directory '/wazuh/src/shared_modules/rsync/build'
    make[4]: Entering directory '/wazuh/src/shared_modules/rsync/build'
    [ 87%] Building CXX object testtool/CMakeFiles/rsync_test_tool.dir/main.cpp.o
    [ 87%] Building CXX object testtool/CMakeFiles/rsync_test_tool.dir/agentEmulator.cpp.o
    [ 87%] Building CXX object testtool/CMakeFiles/rsync_test_tool.dir/oneTimeSync.cpp.o
    [ 87%] Building CXX object testtool/CMakeFiles/rsync_test_tool.dir/managerEmulator.cpp.o
    [100%] Linking CXX executable ../bin/rsync_test_tool
    make[4]: Leaving directory '/wazuh/src/shared_modules/rsync/build'
    [100%] Built target rsync_test_tool
    make[3]: Leaving directory '/wazuh/src/shared_modules/rsync/build'
    make[2]: Leaving directory '/wazuh/src/shared_modules/rsync/build'
    make[1]: Leaving directory '/wazuh/src'
    make: *** [Makefile:783: server] Error 2
     Error 0x5.
     Building error. Unable to finish the installation.

    PLAY RECAP ******************************************************************************************************************************************************
    localhost                  : ok=7    changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
    wazuh-master               : ok=4    changed=3    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

</details>

### Expected behavior

The provisioning phase of each environment used for the system tests should not raise any error once the base Docker image is changed to one containing a GCC version of `9.3.0` or another compatible with the Wazuh `4.7.0` code.

**Comment by Rebits** [comment]
Currently working on the creation of a custom Docker image to run system tests

**Comment by Rebits** [comment]
Changed image for system tests. During the manager installation it has been detected that some wazuh process were in dead state and the Stop manager task is never ending.
```
   9610 ?        SN     0:00  \_ /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
   9613 ?        SN     0:00  \_ /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
   9616 ?        SN     0:00  \_ /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh-apid.py
   8961 ?        ZNs    0:00 [wazuh-csyslogd] <defunct>
   8962 ?        ZN     0:00 [wazuh-csyslogd] <defunct>
   8967 ?        ZNs    0:00 [wazuh-dbd] <defunct>
   8968 ?        ZN     0:00 [wazuh-dbd] <defunct>
   8977 ?        ZNs    0:00 [wazuh-agentless] <defunct>
   8978 ?        ZN     0:00 [wazuh-agentless] <defunct>
   8991 ?        ZNs    0:00 [wazuh-authd] <defunct>
   8992 ?        SNl    0:00 /var/ossec/bin/wazuh-authd
   9004 ?        ZNs    0:00 [wazuh-db] <defunct>
   9005 ?        SNl    0:00 /var/ossec/bin/wazuh-db
   9028 ?        ZNs    0:00 [wazuh-execd] <defunct>
   9029 ?        SNl    0:00 /var/ossec/bin/wazuh-execd
   9039 ?        ZNs    0:00 [wazuh-analysisd] <defunct>
   9040 ?        SNl    0:01 /var/ossec/bin/wazuh-analysisd
   9048 ?        ZNs    0:00 [wazuh-syscheckd] <defunct>
   9049 ?        SNl    0:00 /var/ossec/bin/wazuh-syscheckd
   9060 ?        ZNs    0:00 [wazuh-remoted] <defunct>
   9061 ?        ZN     0:00 [wazuh-remoted] <defunct>
   9062 ?        SNl    0:00 /var/ossec/bin/wazuh-remoted
   9094 ?        ZNs    0:00 [wazuh-logcollec] <defunct>
   9095 ?        SNl    0:00 /var/ossec/bin/wazuh-logcollector
   9111 ?        ZNs    0:00 [wazuh-monitord] <defunct>
   9112 ?        SNl    0:00 /var/ossec/bin/wazuh-monitord
   9120 ?        ZNs    0:00 [wazuh-modulesd] <defunct>
   9121 ?        ZN     0:00 [wazuh-modulesd] <defunct>

```

**Comment by Rebits** [comment]
After conducting some research, it has come to our attention that the reported [issue](https://github.com/wazuh/wazuh-qa/issues/4592#issuecomment-1776801409) exclusively manifests when the manager is initiated either through an exec shell or via an Ansible task.

For additional details, please refer to the following link: https://github.com/wazuh/wazuh/issues/19829

Regrettably, due to this reported issue, resolving the problem has become challenging. The presence of dead state processes significantly hampers the termination process, potentially disrupting certain tests.

**Comment by Rebits** [comment]
Continue issue changing base image to `dontpanicat/debian`

**Comment by pro-akim** [comment]
## Review Notes:

Due to the large size of the new containers, big_40_agents_cluster, the test will need to be run with packages.

**Comment by Rebits** [comment]
### Big 40 agents cluster failure

Deployment fails due to the lack of space to perform the provision. 
This could be related to the increasal size of the virtual size of the new image
```
80f92af5104e   wazuh/ubuntu_jammy_system_tests:1     "/panic/entry tail -…"   21 minutes ago   Up 21 minutes                                                                                 wazuh-agent1        31MB (virtual 9.19GB)
```

However it is important to notice that this Virtual size is used only once irregardless of how many containers are started from the same image. So, this should not lead into a unstable environment if we increase the resources of the host.
In addition it is proposed to delete the` /wazuh` content in all the agents, in order to decrease the required resources of the environment

### Manager agent

Fixed in https://github.com/wazuh/wazuh-qa/pull/4655/commits/5d9e29732ef70bfa70aea65c785b7b882510e115

**Comment by pro-akim** [comment]
##Update 
Moved back to pending review, due to changes in the package setting

**Comment by pro-akim** [comment]
## Review Update
- enrollment_cluster :red_circle: the test showed fails but in the test, not in the process of creating the test environment (irrelevant error in relation with this issue)
- manager_agent :green_circle: test passed

Moved to final pending review

**Comment by pro-akim** [comment]
## Review Notes Upgrade

Tested enrollment_cluster with wazuh_qa 4.6.0 but with new images

Same errors were found:
https://github.com/wazuh/wazuh-qa/files/13236512/enrollment.4.6.0.zip

**Comment by pro-akim** [comment]
## Review Notes Upgrade
Tested enrollment with wazuh 4.6.0 and new docker images

Still, the same errors were detected.

[enrollment 4.6.0.zip](https://github.com/wazuh/wazuh-qa/files/13236512/enrollment.4.6.0.zip)

A new issue is opened to fix the enrollment cluster test.

Moved to pending final review

**Comment by pro-akim** [comment]
## Review Notes 
Reviewing different environments:       
      
|Environment|Result|Report|Note|
|---|---|---|---|
|agentless_cluster|:green_circle:| [agentless_cluster.zip](https://github.com/wazuh/wazuh-qa/files/13206664/agentless_cluster.zip)
|basic_cluster|:green_circle:| [basic_cluster.zip](https://github.com/wazuh/wazuh-qa/files/13215205/basic_cluster.zip)
|basic_environment|:green_circle:|[basic_environment.zip](https://github.com/wazuh/wazuh-qa/files/13216210/basic_environment.zip)
|big_cluster_40_agents|:green_circle:| [big_40.zip](https://github.com/wazuh/wazuh-qa/files/13221282/big_40.zip) |Size of new containers take too much space to create an infrastructure of 40 agents, but running it with packages works
|enrollment_cluster|:green_circle:|[enrollment_cluster.zip](https://github.com/wazuh/wazuh-qa/files/13226436/enrollment_cluster.zip) | fixes done [enrollment_cluster_fixed.zip](https://github.com/wazuh/wazuh-qa/files/13238723/enrollment_cluster_fixed.zip)
agerAgent.zip)
|four_manager_disconnected_node|:green_circle:| [four_manager.zip](https://github.com/wazuh/wazuh-qa/files/13219768/four_manager.zip)
|manager_agent|:green_circle:|https://github.com/wazuh/wazuh-qa/files/13224842/manager_agent.zip | 
|one_manager_agent|:green_circle:| [one_manager_agent.zip](https://github.com/wazuh/wazuh-qa/files/13205408/OneMan





|Status|Description|
|---|---|
|:black_circle:|Available|
|:yellow_circle:|In progress|
|:green_circle:|Passed|
|:red_circle:|Failed|
|:white_circle:|Skipped|
 

LGTM

**Comment by davidjiglesias** [comment]
LGTM!
